pkg/util/utils_test.go

```diff
@@ -57,5 +60,47 @@ func TestRetry(t *testing.T) {
 	if err := Retry(4, f); err == nil {
 		t.Fatalf("Error should have been raised by retry.")
 	}
+}
+
+func errEquals(err1 error, err2 error) bool {
```

Comment:
I think checking the actual error text is a little overkill. It would make the code brittle if we ever change the message or improve error handling. You can drop this and replace it with just a bool indicating whether an error was thrown.